lib/Dialect/FIRRTL/Transforms/ExpandWhens.cpp

```diff
@@ -551,24 +551,82 @@ class WhenOpVisitor : public LastConnectResolver<WhenOpVisitor> {
 
   /// Concurrent and of a property with the current condition.  If we are in
   /// the outer scope, i.e. not in a WhenOp region, then there is no condition.
-  Value ltlAndWithCondition(Operation *op, Value value) {
-    // 'ltl.and' the value with the current condition.
-    return OpBuilder(op).createOrFold<LTLAndIntrinsicOp>(
-        condition.getLoc(), condition.getType(), condition, value);
+  Value ltlAndWithCondition(Operation *op, Value property) {
+    // Look through nodes.
+    while (auto nodeOp = property.getDefiningOp<NodeOp>())
+      property = nodeOp.getInput();
+
+    // Look through `ltl.clock` ops.
+    if (auto clockOp = property.getDefiningOp<LTLClockIntrinsicOp>()) {
```

Comment:
shouldn't this `if` be a `while`? I forget if we somewhere enforce the rule that you can't have 
```
%0 = ltl.clock ...
%1 = ltl.clock %1
...
```
because in that case this still wouldn't fold correctly no?

Reply:
This should work also in that case, since there's a recursive call on the line just below. So the condition would be pushed all the way through any clocks and `firrtl.node`s :slightly_smiling_face: